README.md

```diff
@@ -156,16 +156,40 @@ a checksum for you. Note that checksumming only works for downloading files,
 not directories, but checksumming will work for any protocol.
 
 To checksum a file, append a `checksum` query parameter to the URL.
-The paramter value should be in the format of `type:value`, where
-type is "md5", "sha1", "sha256", or "sha512". The "value" should be
-the actual checksum value. go-getter will parse out this query parameter
-automatically and use it to verify the checksum. An example URL
-is shown below:
+The parameter value should be in the format of `type:value`, where
+type is "md5", "sha1", "sha256", "sha512" or "file". The "value" should be
+the actual checksum value or download URL for "file".
+When `type` part is omitted it will be guessed based on the length
+of the checksum string.
+go-getter will parse out this query parameter automatically
+and use it to verify the checksum. 
+Examples:
 
 ```
 ./foo.txt?checksum=md5:b7d96c89d09d9e204f5fedc4d5d55b21
 ```
 
+```
+./foo.txt?checksum=b7d96c89d09d9e204f5fedc4d5d55b21
+```
+
+```
+./foo.txt?checksum=file:./foo.txt.sha256sum
+```
+ 
+When checksumming from a file - ex: with `checksum=file:url` - go-getter will
+get the file linked in the URL after `file:` using a new getter
+(`getter.GetFile`). For example, in
```

Comment:
I was expecting this to be the same getter? The below example would indicate this, and especially if this was a remote source the HTTP getter may need to have the same client and headers that are in the getter used to download the target.

Reply:
Gotcha; will make the checksum func a client func !

Reply:
Hmm, just looking at 487c31a, I think you'd just do better with adding the client as another call parameter.

Everything doesn't necessarily need to be an object, and in this case seeing as you only need it in a few places in checksumFromFile, you could probably just deal with the slight unsighlyness of having an extra parameter versus the risk of making several possibly breaking changes throughout the codebase.

Reply:
I totally agree with you here; I just took that path to avoid having to have to do something like as this feel not idiomatic to me :
```go
oldSrc := client.Src
client.Src = checksumFile
cliend.Get()
client.Src = oldSrc
//...
```

or

```go
checksumClient := *client // copy client
checksumClient.Src = checksumFile
checksumClient.Get()
//...
```

I think I will go for the second option then 🙂

Reply:
Yeah, I keep forgetting about how `Get` is not paramterized on source and destination. So this is partially my bad, sorry about that!

Maybe instead of the full client then, how about just passing the getters?

Reply:
No worries, I totally skipped over using getters without any explanation 🙂. I shouldn't have.
Mainly because if I do this I need to re-implement part of what the client does and I thought it was better to just have a breaking change instead. To avoid future issues like "getting a checksum file in my special case doesn't work".

But to get the checksum file I need to:
* guess the getter based on url
* then go-get it

The rest of the client features I could think of are not necessary though :
* checksumming a checksum sounds weird ( this cannot be achieved right now because checksum branching case is in client )
* I have never seen archived/git-stored checksum files
-> checksums are files in general

I will go for passing the getters around then, I'm almost tempted to make a `Getters struct` type 🤔 🙂 

Thanks for your patience !